src/services/data_model_engine.js

```diff
@@ -213,9 +211,9 @@ export default class DataModelEngine {
     };
     for (const waitingBundle of waitingBundles) {
       try {
-        const {blockNumber, transactionHash} = await this.uploadRepository.uploadBundle(waitingBundle.bundleId, waitingBundle.storagePeriods);
+        const {blockNumber, transactionHash, uploadResult} = await this.uploadRepository.ensureBundleIsUploaded(waitingBundle.bundleId, waitingBundle.storagePeriods);
         await this.entityRepository.storeBundleProofMetadata(waitingBundle.bundleId, blockNumber, transactionHash);
-        summary.ok.push(waitingBundle.bundleId);
+        summary.ok.push({bundleId:waitingBundle.bundleId, uploadResult});
```

Comment:
for consistency with `summary.failed` I would move `summary.ok` from being an array to being a map;